pkg/server/server_authz_test.go

```diff
@@ -615,6 +700,91 @@ func TestCheckAuthz(t *testing.T) {
 	})
 }
 
+func TestCheckWriteAuthz(t *testing.T) {
+	t.Cleanup(func() {
+		goleak.VerifyNone(t)
+	})
+	ds := memory.New()
+	t.Cleanup(ds.Close)
+
+	model := &openfgav1.AuthorizationModel{
+		SchemaVersion:   typesystem.SchemaVersion1_1,
+		TypeDefinitions: testStoreModelWithModule(),
+	}
+	typesys := typesystem.New(model)
+
+	t.Run("checkWriteAuthz_no_authz", func(t *testing.T) {
+		openfga := MustNewServerWithOpts(
+			WithDatastore(ds),
+		)
+		t.Cleanup(openfga.Close)
+
+		err := openfga.checkWriteAuthz(context.Background(), &openfgav1.WriteRequest{
+			StoreId: "store-id",
+			Deletes: &openfgav1.WriteRequestDeletes{
+				TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+					{Object: "folder-with-module:2", Relation: "viewer", User: "user:jon"},
+				},
+			},
+		}, typesys)
+		require.NoError(t, err)
+	})
+
+	t.Run("checkWriteAuthz_with_authz", func(t *testing.T) {
+		openfga := MustNewServerWithOpts(
+			WithDatastore(ds),
+		)
+		t.Cleanup(openfga.Close)
+
+		clientID := "validclientid"
+		settings := newSetupAuthzModelAndTuples(t, openfga, clientID)
+
+		openfga.authorizer = authz.NewAuthorizer(&authz.Config{StoreID: settings.rootData.id, ModelID: settings.rootData.modelID}, openfga, openfga.logger)
+
+		t.Run("error_when_GetModulesForWriteRequest_errors", func(t *testing.T) {
+			err := openfga.checkWriteAuthz(context.Background(), &openfgav1.WriteRequest{
+				StoreId: settings.testData.id,
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{Object: "unknown:2", Relation: "viewer", User: "user:jon"},
+					},
+				},
+			}, typesys)
+			require.Error(t, err)
+		})
+
+		t.Run("error_when_checkAuthz_errors", func(t *testing.T) {
+			ctx := authclaims.ContextWithAuthClaims(context.Background(), &authclaims.AuthClaims{ClientID: clientID})
+			err := openfga.checkWriteAuthz(ctx, &openfgav1.WriteRequest{
+				StoreId: settings.testData.id,
+				Deletes: &openfgav1.WriteRequestDeletes{
+					TupleKeys: []*openfgav1.TupleKeyWithoutCondition{
+						{Object: "workspace:2", Relation: "guest", User: "user:jon"},
+					},
+				},
+			}, typesys)
+			require.Error(t, err)
+			require.Equal(t, "rpc error: code = Code(403) desc = the principal is not authorized to perform the action", err.Error())
```

Comment:
nitpik (non-blocking): You don't need to do two assert.  You can just call

require.Equal(t, status.Error(codes.Code..... ), err).